kernel/src/hil/aes.rs

```diff
@@ -0,0 +1,14 @@
+use returncode::ReturnCode;
+
+pub trait AESDriver {
+    fn init(&self);
+    fn set_key(&self, key: &'static mut [u8], len: u8);
```

Comment:
Ok, I'm not sure if I understand the compile-time checking either because the userapp can compiled and loaded at later stage. 
But maybe you can ensure that once loading the userapp? If that's the case then we are fine.
However I will start to clean-up to code and move static mut stuff in to Cell/MapCell arrays, rename stuff, adding more comments and so on.